services/qs-spark-service/src/main/java/ai/quantumics/api/controller/AwsConnectionController.java

```diff
@@ -0,0 +1,82 @@
+package ai.quantumics.api.controller;
+
+import ai.quantumics.api.helper.ControllerHelper;
+import ai.quantumics.api.model.AWSDatasource;
+import ai.quantumics.api.model.Projects;
+import ai.quantumics.api.model.QsUserV2;
+import ai.quantumics.api.req.AwsDatasourceRequest;
+import ai.quantumics.api.service.AwsConnectionService;
+import ai.quantumics.api.util.DbSessionUtil;
+import ai.quantumics.api.util.ValidatorUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1/connection")
+public class AwsConnectionController {
+    private final AwsConnectionService awsConnectionService;
+    private final DbSessionUtil dbUtil;
+    private final ValidatorUtils validatorUtils;
+
+    private final ControllerHelper controllerHelper;
+
+    public AwsConnectionController(AwsConnectionService awsConnectionService, DbSessionUtil dbUtil,
+                                   ValidatorUtils validatorUtils, ControllerHelper controllerHelper) {
+        this.awsConnectionService = awsConnectionService;
+        this.dbUtil = dbUtil;
+        this.validatorUtils = validatorUtils;
+        this.controllerHelper = controllerHelper;
+    }
+
+    @PostMapping("/saveConnection")
+    public ResponseEntity<AWSDatasource> saveConnection(@RequestBody AwsDatasourceRequest awsDatasourceRequest)
+            throws Exception {
+
+        dbUtil.changeSchema("public");
+        QsUserV2 user = validatorUtils.checkUser(awsDatasourceRequest.getUserId());
```

Comment:
if awsDatasourceRequest is null? if user is null? if project is null?.  it is applicable to all places in this file

Reply:
Added validations.